vmware-base/src/main/java/com/cloud/hypervisor/vmware/mo/HypervisorHostHelper.java

```diff
@@ -839,10 +849,79 @@ private static void createPortGroup(String physicalNetwork, String networkName,
         }
     }
 
-    public static boolean isSpecMatch(DVPortgroupConfigInfo currentDvPortgroupInfo, DVPortgroupConfigSpec newDvPortGroupSpec) {
+    private static boolean eitherObjectNull(Object obj1, Object obj2) {
+        return (obj1 == null && obj2 != null) || (obj1 != null && obj2 == null);
+    }
+
+    private static boolean areBoolPoliciesDifferent(BoolPolicy currentPolicy, BoolPolicy newPolicy) {
+        return eitherObjectNull(currentPolicy, newPolicy) ||
+                (newPolicy != null && newPolicy.isValue() != currentPolicy.isValue());
```

Comment:
extra not null checks may not be required (both objs are not null ensured before using _eitherObjectNull()_), please check wherever applicable.

Reply:
There is no check for both objects null. `eitherObjectNull()` checks if only one of them is null.

Reply:
correct, so when both are null, can be a NPE?

Reply:
@sureshanaparti 
Four cases:

1. currentPolicy is null, newPolicy is not null       <- Will be handled by `eitherObjectNull`
2. currentPolicy is not null, newPolicy is null       <- Will be handled by `eitherObjectNull`
3. currentPolicy is null, newPolicy is null             <- Will fail `newPolicy != null` check in second part of the condition
4. currentPolicy is not null, newPolicy is not null <- Should be handled by `newPolicy.isValue() != currentPolicy.isValue()`

Actually I added null check of both vars initially but IntelliJ suggested simplification and I guess it was right

Reply:
@shwstppr why not you handle case 3) in the private method itself?

Reply:
@sureshanaparti which private method?
If both objects are null, `areBoolPoliciesDifferent` method should return false. That is why